tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/ManualInstrumentation/Tracer/StartActiveImplementationIntegration.cs

```diff
@@ -29,13 +30,22 @@ namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.ManualInstrumentation.Tr
 [EditorBrowsable(EditorBrowsableState.Never)]
 public class StartActiveImplementationIntegration
 {
+    private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor<StartActiveImplementationIntegration>();
+
     internal static CallTargetState OnMethodBegin<TTarget, TSpanContext>(TTarget instance, string operationName, TSpanContext? parent, string? serviceName, DateTimeOffset? startTime, bool? finishOnClose)
         where TTarget : ITracerProxy
     {
         // parent should _normally_ be a manual span, unless they've created a "custom" ISpanContext
         var parentContext = SpanContextHelper.GetContext(parent);
 
-        var tracer = (Datadog.Trace.Tracer)instance.AutomaticTracer;
+        if (instance.AutomaticTracer is not Datadog.Trace.Tracer tracer)
+        {
+            Log.Error(
+                "Error: instance.AutomaticTracer is not a Datadog.Trace.Tracer: {TracerType}. This should never happen, and indicates a problem with automatic instrumentation.",
```

Comment:
If you log an error in StartActive, shouldn't you log on in StartSpan as well?

Reply:
Yeah, potentially. It's only called by the OpenTracing implementation, which is now obsolete, so I was undecided, but why not. Done in https://github.com/DataDog/dd-trace-dotnet/pull/6169/commits/8e957bbee5d0f407c0a9a4e97ed5471945d369a3